index.sh

```diff
@@ -12,31 +12,35 @@ set -e
 
 # Arguments:
 # 1 - folder to search files in
-function buildIndex() {
-    metaInfoFields=('displayName' 'description' 'tags' 'icon' 'globalMemoryLimit')
+if [[ ! $1 ]]; then echo "Error: must specify path to devfiles, eg., $0 /path/to/che-devfile-registry/devfiles"; exit 1; fi
+devfilesDir=$1; devfilesDir=${devfilesDir%/} # trim trailing slash if present
+
+metaInfoFields=('displayName' 'description' 'tags' 'icon' 'globalMemoryLimit')
 
-    ## search for all devfiles
-    readarray -d '' arr < <(find "$1" -name 'meta.yaml' -print0)
+function buildIndex() {
 
     FIRST_LINE=true
     echo "["
 
-    ## now loop through meta.yaml files
-    for i in "${arr[@]}"
-    do
-        if [ "$FIRST_LINE" = true ] ; then
+    ## loop through meta.yaml files
+    for i in $(ls -1 ${devfilesDir}/*/meta.yaml | sort); do
```

Comment:
It seems shellcheck is reporting some warnings

Reply:
can you tell me how/where to run shellcheck and what the warning(s) are?

Reply:
copy paste into https://www.shellcheck.net/

And there is a fixme button

Reply:
ok, I can do some replacements for cat | grep --> grep and wrap some quotes around stuff.

But replacing `ls ` with `find` means the container needs to install `find`, and assuming that the devfiles folders DON'T contain spaces, there's no need to use find here.